Add new terms for glossary: RHEL software management (#261)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc

*See also*: xref:distinguished-name[distinguished name]

[discrete]
[[baseos-repository]]
==== image:images/yes.png[yes] BaseOS repository (noun)

> *Description*: In Red Hat Enterprise Linux, the  _BaseOS repository_ contains the core set of the underlying OS functionality that provides the foundation for all installations.

Reviewer: ```suggestion
*Description*: While installing RHEL, the _BaseOS repository_ provides the core operating system functions as RPM packages. 
```

Author: As per discussion with SME leaving the original version.

> *Description*: In Red Hat Enterprise Linux, the  _BaseOS repository_ contains the core set of the underlying OS functionality that provides the foundation for all installations.

Reviewer 2: Agree with Rolfe that there's something about "core set of ...  functionality" that doesn't quite jibe.

Author: As per the discussion with SME, leaving the original version.

*See also*: xref:distinguished-name[distinguished name]

[discrete]
[[baseos-repository]]
==== image:images/yes.png[yes] BaseOS repository (noun)

> *Description*: RHEL content in the _BaseOS repository_ provides the core set of the underlying OS functionality that provides the foundation for all installations.

Reviewer 3: Since this seems pretty specific to RHEL, should probably reword to start with "In Red Hat Enterprise Linux,"

I see you did do that with a few other terms, so not sure if it's not specific to RHEL. But maybe do another quick look throughout the PR to start with that for all that are pretty specific to only RHEL?

> *Description*: RHEL content in the _BaseOS repository_ is intended to provide the core set of the underlying OS functionality that provides the foundation for all installations.

Reviewer 4: ```suggestion
*Description*: RHEL content in the _BaseOS repository_ provides the core set of the underlying OS functionality that provides the foundation for all installations.
```

Just a little suggestion here because the passive 'is intended to' doesn't seem to add anything meaningful. It's not clear who is the one who intends. If you keep it though, maybe it'd be better to explain it a bit more.

*See also*:

[discrete]
[[binary-rpm]]
==== image:images/yes.png[yes] binary RPM (noun)

> *Description*: A _binary RPM_ is an RPM package that contains the binaries built from sources and patches.

Reviewer: ```suggestion
*Description*: A _binary RPM_ is a software installation package that contains a compiled binary executable file.
```

*See also*: xref:filestore[FileStore], xref:object-store[Object Store]

[[boolean-dependencies]]
==== image:images/yes.png[yes] Boolean dependencies (noun)

> *Description*: In Red Hat Enterprise Linux, _Boolean dependencies_ are Boolean expressions such as `if`, `and`, `or`, and other expressions that are used in the `Requires`, `Conflicts`, and `Weak` dependency directives. Boolean dependencies are also known as _Rich dependencies_.

Reviewer 2: definitely "Rich," capital R? i'm finding both in search results (ofc).

Author: Yes, already discussed in other comment above

*See also*: xref:filestore[FileStore], xref:object-store[Object Store]

[discrete]
[[boolean-dependencies]]
==== image:images/yes.png[yes] Boolean dependencies (noun)

> *Description*: _Boolean dependencies_ are Boolean expressions (if, and, or, and other expressions) used in the Requires, Conflicts, and Weak dependency directives. Boolean dependencies are also known as Rich dependencies.

Reviewer 5: ```suggestion
*Description*: _Boolean dependencies_ are Boolean expressions (`if`, `and`, `or`, and other expressions) used in the Requires, Conflicts, and Weak dependency directives. Boolean dependencies are also known as _Rich dependencies_.
```
- I think these are literals and should be in back ticks.
- I'm not sure, but should `Requires`, `Conflicts`, and `Weak` also be in back ticks?
- `Rich dependencies` should also be italicized, I believe.

Reviewer 6: +1
Thanks for catching that.

*See also*: xref:filestore[FileStore], xref:object-store[Object Store]

[discrete]
[[boolean-dependencies]]
==== image:images/yes.png[yes] Boolean dependencies (noun)

> *Description*: _Boolean dependencies_ are Boolean expressions (`if`, `and`, `or`, and other expressions) used in the Requires, Conflicts, and Weak dependency directives. Boolean dependencies are also known as _Rich dependencies_.

Author: Thank you, I checked it with the writer.

*Incorrect forms*: byte code

*See also*:

[discrete]
[[byte-compiled-program]]
==== image:images/yes.png[yes] byte-compiled program (noun)

> *Description*: _Byte-compiled programs_ are programs that need to be compiled into byte code, which is then executed by a language virtual machine.

Reviewer 2: --> ... are programs that must be compiled into byte code

change "executed" to "started"?

Reviewer: IBM Style: "Execute: Use only if a simpler term, such as run, issue, start, or enter is inappropriate."

Reviewer: ```suggestion
*Description*: _Byte-compiled programs_ are programs that need to be compiled into byte code before running in a language virtual machine.
```

Reviewer: ```suggestion
*Description*: _Byte-compiled programs_ are programs that must be compiled into byte code before they can run in a language virtual machine.
```
